shell/browser/ui/electron_desktop_window_tree_host_linux.cc

```diff
@@ -46,13 +48,53 @@ void ElectronDesktopWindowTreeHostLinux::OnBoundsChanged(
     const BoundsChange& change) {
   views::DesktopWindowTreeHostLinux::OnBoundsChanged(change);
   UpdateFrameHints();
+  // The OnWindowStateChanged should receive all updates but currently it
+  // doesn't receive changes to fullscreen status, so we catch them here.
+  UpdateWindowState(platform_window()->GetPlatformWindowState());
```

Comment:
I tested the branch and this only seems to affect X11 platform, stems from [synchronously handling the fullscreen states](https://source.chromium.org/chromium/chromium/src/+/main:ui/ozone/platform/x11/x11_window.cc;l=592-605) and ends up not firing the [state change](https://source.chromium.org/chromium/chromium/src/+/main:ui/ozone/platform/x11/x11_window.cc;l=1353-1357).

Atleast in my testing wayland platform does not suffer from this, can we guard this behind `electron_can_call_x11` and also improve the comments around it.

Reply:
Thanks for finding the code responsible! I've guarded code behind check and expanded comment.